### DISCUSSION TITLE
More complete network energy usage information

### DISCUSSION THREAD

**Discussion by not-Ian** [mistakia/nano-community#93] [Proposals]
**Tasks**

- [ ] gather a list of PoW Providers
  - [ ] collect energy usage information
  - [ ] setup a pipeline to collect # of blocks per day
- [ ] allow reps to update hardware info (https://github.com/mistakia/nano-community/issues/105)
- [ ] setup account and block labeling to better understand block production (https://github.com/mistakia/nano-community/discussions/100)

-----

Putting some thoughts down regarding how much energy the network actually uses. I'd like to address the often-repeated talking point that the entire Nano network could run off of a single wind turbine.

-----

Below is a screenshot of a spreadsheet I've been working on. Most of it was taken from taken from the Nano Developer Work Generation Guide: https://docs.nano.org/integration-guides/work-generation/#example-benchmarks. 3090 datapoint added by my own test.

![image](https://github.com/mistakia/nano-community/assets/8951448/bd600090-ae76-4830-8a7f-a6b5dd5187ac)

-----

Example, best-case scenario:

- Nvidia Tesla V100 (most efficient GPU) is doing 100% of the work for the network
- Constant, 24hr send block creation is 7.25 * 60 * 60 * 24 * 365 = 228,636,000 send blocks/year
    - But, we also need receive blocks.
    - There are 31,536,000 seconds in a year,
    - There’s some number of seconds, X, where:
        - 7.25 * (31,536,000 - X) = 420.33 * X
    - multiply the 7.25 out:
        - 228,636,000 - 7.25X = 420.33 * X
    - divide by 420.33:
        - 543,944 - ( ( 7.25 * X ) / 420.33) = X
    - simplify 7.25 / 420.33:
        - 543,944 - ( 0.0172483525 * X ) = X
    - add the X component on the left to the right:
        - 543,944 = 1.0172483525 * X
    - divide by 1.01724…:
        - **534,720.94 = X**
    - Which means:
        - 7.25 * (31,536,000 - 534,720.94) = 224,759,280 send blocks/year
        - 420.33 * 534,720.94 = 224,759,252 receive blocks/year
        - Looks approximately correct ✅
- That uses 250W * 24 * 365 = 2,190,000Wh, or 2,190kWh
- 2,190kWh / 224,759,252 transactions = 0.0000097438 kWh/transaction
    - = 0.0097438 Wh / tx
    - = **9.7438 mWh / tx**

This is JUST PoW, and not energy used by the rest of the network for confirmation.

Also, again, this is a best case scenario. We need to weight it against whatever distribution real-world PoW generation follows.

**This would be an important piece of information we could gather that would legitimize this calculation - what GPUs does the network use? Can we survey the biggest nodes/wallets to see what hardware they use to generate work?**

Though it’s interesting to see the cross over from dedicated GPU to CPU - I doubt the real-world number is worse than, like, 10x this best-case number. 

-----

- USGS says the average wind turbine in the USA:
    - “...At a 42% capacity factor (i.e., the average among recently built wind turbines in the United States, per the 2021 edition of the U.S. Department of Energy’s *Land-Based Wind Market Report*), that average turbine would generate over 843,000 kWh per month...”
    - https://www.usgs.gov/faqs/how-many-homes-can-average-wind-turbine-power
- 843,000 kWh / mo x 12 = 10,116,000 kWh / year

[Nano.community](http://nano.community/) currently lists network TDP for the last 24 hours as 40.08 kWh

Assuming holding steady for a year, that’s 14,600 kWh / year **(Do I have this right? 40.08 kWh _accumulated_ over the last 24 hours, or 40.08 kWh, _per hour_, averaged over the last 24 hours?)**

Taking that into account, we reduce the 10,116,000 kWh produced by the windmill by 14,600 kWh:

- = 10,101,400 kWh / year
- = 10,101,400,000 Wh / year
- = 10,101,400,000,000 mWh / year

10,101,400,000,000 mWh / year / **9.7438 mWh / tx**

= 1,038,198,649,397 tx / year

= **32,873 tx / s (best case PoW generation, and using estimated PR TDP for CPU only)**

**Comment by not-Ian** [comment]
Once we hash this out, I envision an article:

Headline: "No, the Nano network does not run off of a single wind turbine"
Subtitle: "It probably uses much, much less."

**Comment by mistakia** [comment]
I'm not sure how feasible it will be to attribute a specific GPU to a given block (and thus PoW). We would need a way to link GPUs to accounts (and therefore blocks). This could be possible for services but I'm having a hard time figuring out how it could be done for users using various wallets.

I think a crude first attempt could be polling various sources of PoW to get an idea of what GPUs are used and from that information establishing a low, median and high estimate. We can then combine that with block counts by type over X period for an estimate.

**Reply by not-Ian** [reply]
Agreed there's not a clear way to attribute a specific GPU to a given block. Agreed a crude first attempt would include polling various sources of PoW to get an idea of what GPUs are used.

If we could find what percent of regular block traffic each of these PoW creates, we could get close. Say, if Kraken contributes to 5% of regular traffic, we weight whatever PoW source they use against that, and add it to the total. Between Kraken, Binance, Natrium, and other wallets, we don't need to cover 100% of traffic to get an idea.

Is it hopeless to even consider asking Kraken/Binance, or have you contacted them before when you were trying to find CPU TDP?

First draft list of who we need to contact:

- [ ] Kraken
- [ ] Binance
- [x] rpc.nano.to (Cake Wallet, Nautilus, Nault.pro, Nault.cc, 1 undisclosed major wallet(?), 2 undisclosed exchanges)
- [ ] Boompow? (Graham nanotipbot, Natrium?)
- [x] https://rainstorm.city/api (Nault, crypto.com?)
- [ ] https://github.com/Nault/Nault/pull/612 (Nault - yet to be added)
- [ ] https://github.com/Nault/Nault/pull/611 (Nault - yet to be added)

**Reply by not-Ian** [reply]
Bbedward, who maintains Natrium, mentioned boompow: https://github.com/BananoCoin/boompow

keeri, who doesn't maintain Nault, but seems to know about it, says:

> i don't but it sends requests directly to public nodes. some use fpga with solar power, some use dpow

I opened Nault and found it defaults to `https://rainstorm.city/api`. Other options are MyNanoNinja, nanos.cc. MNN and Nanos.cc seem to not be working?

fosse, who maintains CakeWallet, mentions they use rpc.nano.to, but it's configurable to use other options.

**Reply by mistakia** [reply]
I've never attempted to contact exchanges so can't speak to that. I can work on getting the number of daily blocks attributed to exchanges.

The challenge will be attributing/estimating blocks to those other services. Let's first gather the energy usage information of those services then we can come up with ideas on how to assign energy usage to blocks we can't connect to a PoW source.

**Reply by not-Ian** [reply]
From nano.to dev:

My apologies, I cant't provide this information as transparently as I'd like. 

But I can provide some clarification. [RPC.Nano.To](http://rpc.nano.to/) offers RPC and POW to:

- Cake Wallet *
- Nautilus Wallet *
- [Nault.Pro](http://nault.pro/)
- [Nault.CC](http://nault.cc/) *
- 1 undisclosed major wallet *
- 2 undisclosed exchanges

* The RPC source on this service is user configurable, so [RPC.Nano.TO](http://rpc.nano.to/) is not the only source. 

[RPC.Nano.TO](http://rpc.nano.to/) offers x6 Nvidia GTX 1080 GPUs. However some of them auto-shutoff in times of slow demand. 

The entire system is on a UPS (uninterruptible power supply) that reports an average of 450 Watts/h and about 1200 Watts/h peak demand.

**Reply by not-Ian** [reply]
From jaycoxx, who runs rainstorm.city:
- I use a fpga, so it’s 5W and takes between 5 and 20 seconds to generate work
- it’s about 1000 a day though my fpga uses 5W even if it isn’t processing

**Comment by not-Ian** [comment]
Hey @mistakia , not sure if you saw this question in my original post:

> (Do I have this right? 40.08 kWh accumulated over the last 24 hours, or 40.08 kWh, per hour, averaged over the last 24 hours?)

`40.08 kWh` was what nano.community was displaying for `Energy Usage (TDP) (24hr)` at the time.

FWIW, it's a tiny change in the final number, but I want to get this right. If we did multiply 40.08 by 24 and work that out, the final total number of transactions a windmill could support only drops from 32,873 tx/s, to 31,778 tx/s.

I'd be curious to dig into that CPU TDP number more, and figure out a more meaningful way to work it into the equation. I know you said it was a crude estimate based on polling some reps, right? How did that go - you asked 20 nodes and 10 came back and you added those 10 together? I feel like we could take an average CPU TDP across the nodes that provided data, and multiply it out across every PR? Just to estimate, in the example, what the 10 nodes that didn't respond to your survey contribute to total energy usage.

**Reply by mistakia** [reply]
good question — I glossed over that so thanks for bringing my attention to it. Let me know if this does not answer your question: https://github.com/mistakia/nano-community/issues/102
